### PR TITLE
OCPBUGS-20492: highperfhooks: apply cgroup setting to crun cgroup

### PR DESCRIPTION
**Pull request by haircommander**
#### Does this PR introduce a user-facing change?

```release-note
fix a bug where the cgroup crun configures is different than that CRI-O sets load balancing/cpu quota on
```

### CODE REVIEW

File: test/cgroups.bats
Comment (kolyshkin) [inline test/cgroups.bats:223]: Looks like a slash is missing after `container`

Comment (haircommander) [reply test/cgroups.bats]: oops, and that revealed I was testing it wrong!

File: internal/runtimehandlerhooks/default_cpu_load_balance_hooks.go
Comment (rphillips) [inline internal/runtimehandlerhooks/default_cpu_load_balance_hooks.go:40]: nonblocking: This block could be refactored to a function, since the same code is line high_performance_hooks.go:225

Comment (haircommander) [reply internal/runtimehandlerhooks/default_cpu_load_balance_hooks.go]: done!

File: internal/runtimehandlerhooks/high_performance_hooks.go
Comment (kolyshkin) [inline internal/runtimehandlerhooks/high_performance_hooks.go:212]: Instantiating two cgroup managers for the sake of getting the path to cpuset cgroup seems like an overkill.

I would do it differently -- get the path to cgroup* , and in case its last component is "container", write to the parent cgroup as well.

* as for getting the path to cgroup, you can do it in the same way like it's done before this patch, OR maybe you can actually cache the cgroup path in Container struct somehow (similar to how you do it for Sandbox and CgroupParent).

**Review by kolyshkin — CHANGES_REQUESTED** [review]
I don't like using two instances of cgroup manager to merely get cgroup path.

File: internal/runtimehandlerhooks/high_performance_hooks.go
Comment (MarSik) [reply internal/runtimehandlerhooks/high_performance_hooks.go:212]: I agree the managers are heavy and complicate things. However hardcoding a name based detection is ugly and not future proof.

Comment (haircommander) [reply internal/runtimehandlerhooks/high_performance_hooks.go:212]: > you can do it in the same way like it's done before this patch

this won't work for the sched_load_balance case because we need to disable after the container has stopped too (in case it has load balancing while  it was running but we need to reallocate the cpus it used to use later), which means the proc entry is gone

**Comment by haircommander** [comment]
/hold

this may not be working correctly

**Comment by haircommander** [comment]
/hold cancel

this also needs https://github.com/containers/crun/pull/1332 but the crio pieces are ready

**Comment by haircommander** [comment]
@cri-o/cri-o-maintainers PTAL

**Pull request merged**
